src/contexts/StateContextProvider.tsx

```diff
@@ -416,13 +420,15 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
         );
         mainWorkerRef.current.onmessage = handleMainWorkerResp;
         workerPostReq(mainWorkerRef.current, WORKER_REQ_CODE.LOAD_FILE, {
-            fileSrc: fileSrc,
-            pageSize: getConfig(CONFIG_KEY.PAGE_SIZE),
             cursor: cursor,
             decoderOptions: getConfig(CONFIG_KEY.DECODER_OPTIONS),
+            fileSrc: fileSrc,
+            isPrettified: isPrettified ?? false,
+            pageSize: getConfig(CONFIG_KEY.PAGE_SIZE),
         });
     }, [
         handleMainWorkerResp,
+        isPrettified,
     ]);
```

Comment:
to avoid changing the definition of `loadFile` on changes of `isPrettified` (which may unintentionally cause the file to reload on changes of `isPrettified`):
```suggestion
            isPrettified: isPrettifiedRef.current,
            pageSize: getConfig(CONFIG_KEY.PAGE_SIZE),
        });
    }, [isPrettified]);
```

Reply:
Fixed! I believe this is a typo, the depency should be `[handleMainWorkerResp]`.